ipyparallel/controller/hub.py

```diff
@@ -833,15 +833,26 @@ def mia_task_request(self, idents, msg):
 
     #--------------------- IOPub Traffic ------------------------------
 
-    def save_iopub_message(self, topics, msg):
-        """save an iopub message into the db"""
-        # print (topics)
+    def monitor_iopub_message(self, topics, msg):
+        '''intercept iopub traffic so events can be acted upon'''
```

Comment:
I think it's fine to do this here. The alternative is to do it when the `shutdown_request` comes on the mux channel, but that could potentially cause some issues as there can be results and tasks that are still running and producing replies in between the shutdown request and the engine handling it.